### DISCUSSION TITLE
Is there an api available to just fetch the sources for a particular package?

### DISCUSSION THREAD

**Discussion by smac89** [xmake-io/xmake#4912] [Q&A] [answered]
Is it possible to fetch the sources of a particular version of a package, without building it? For example, this [package](https://github.com/xmake-io/xmake-repo/blob/dev/packages/i/imgui/xmake.lua) has a number of versions listed. Is there anything builtin that allows us to say, that we just want to grab the source of a particular version?

**Comment by waruqi** [comment, marked as answer]
no way now, you need to install it first. but you can use set_base to override `on_install`/`on_test`. it will only download sources.

```
package("zlibsrc")
    set_base("zlib")
    on_install(function() end)
    on_test(function() end)
```

**Reply by smac89** [reply]
Thanks a lot. I put this in a local repo to use it. Works pretty well

**Reply by smac89** [reply]
One more question: Is there a way to cause the source always be cached?

**Reply by waruqi** [reply]
Source tar balls are cached by default and cleaned once a month.

The source directories of release package are cleaned up and the debug packages are retained.
